render/drm_format_set.c

```diff
@@ -76,15 +76,16 @@ bool wlr_drm_format_set_add(struct wlr_drm_format_set *set, uint32_t format,
 
 		if (fmt->len == fmt->cap) {
 			size_t cap = fmt->cap ? fmt->cap * 2 : 4;
-
-			fmt = realloc(fmt, sizeof(*fmt) + sizeof(fmt->modifiers[0]) * cap);
```

Comment:
I don't understand why this change is necessary.

Reply:
`fmt` leaks memory if the reallocation fails, because it will be overwritten with `NULL`. This patch ensures that we keep the old data in place if the reallocation fails.

Reply:
`fmt` is a local variable, so it doesn't matter right?

Reply:
You’re right.